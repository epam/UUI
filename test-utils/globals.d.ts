declare module "jsurl";
declare module "query-string";
