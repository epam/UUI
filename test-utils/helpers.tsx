import React, { ComponentType, ReactElement, ReactNode } from 'react';
import { mount } from 'enzyme';
import { renderHook } from '@testing-library/react-hooks';
import { act as reactAct } from 'react-dom/test-utils';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { ContextProvider, UuiContexts } from '@epam/uui-core';

export const delay = (ms: number = 1): Promise<void> =>
    new Promise(resolve => {
        setTimeout(resolve, ms);
    });

export const testSvc = {} as UuiContexts;

export async function mountHookWithContext<TProps, TResult>(hook: (props: TProps) => TResult, initialProps?: TProps) {
    const wrapper = ({ children }: { children?: React.ReactNode }) => (
        <ContextProvider onInitCompleted={svc => Object.assign(testSvc, svc)}>{children}</ContextProvider>
    );

    const { waitForNextUpdate, rerender, ...rest } = renderHook<TProps, TResult>(hook, { wrapper, initialProps: initialProps });

    await waitForNextUpdate();

    return {
        rerender: (props: TProps) => rerender({ ...props, children: undefined }),
        waitForNextUpdate,
        ...rest,
    };
}

export const mountWithContextAsync = async (children: ReactNode, enableLegacyContext = false) => {
    const wrapper = mount(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {children}
        </ContextProvider>
    );

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const mountWrappedComponentAsync = async <TComponent extends ComponentType<TProps>, TProps>(
    Component: TComponent,
    props: TProps,
    enableLegacyContext = false
) => {
    const renderComponent = (properties: TProps) =>
        mount(
            React.createElement(
                (p: any) => (
                    <ContextProvider
                        onInitCompleted={svc => {
                            Object.assign(testSvc, svc);
                        }}
                        enableLegacyContext={enableLegacyContext}
                    >
                        <Component {...p} />
                    </ContextProvider>
                ),
                properties
            )
        );
    const wrapper = renderComponent(props);

    await reactAct(delay);
    wrapper.update();

    return wrapper;
};

export const renderWithContextAsync = async (component: ReactElement, enableLegacyContext = false) => {
    const result = renderer.create(
        <ContextProvider
            onInitCompleted={svc => {
                Object.assign(testSvc, svc);
            }}
            enableLegacyContext={enableLegacyContext}
        >
            {component}
        </ContextProvider>
    );

    await rendererAct(delay);
    return result.toJSON();
};
//
// global = {
//     ...global,
//     delay,
//     testSvc,
//     mountWithContextAsync,
//     mountWrappedComponentAsync,
//     renderWithContextAsync,
// } as any;

// // @ts-ignore
// global.delay = delay;
// // @ts-ignore
// global.testSvc = testSvc;
// // @ts-ignore
// global.mountWithContextAsync = mountWithContextAsync;
// // @ts-ignore
// global.mountWrappedComponentAsync = mountWrappedComponentAsync;
// // @ts-ignore
// global.renderWithContextAsync = renderWithContextAsync;
