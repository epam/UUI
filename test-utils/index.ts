export * from "./src/renderingHelpers";
export * from "./src/jestHelpers";
export * from "./src/setupComponentForTest";


