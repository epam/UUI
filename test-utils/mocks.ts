export const windowMock = {
    matchMedia: () => ({matches: false}),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    document: {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
    },
} as any;