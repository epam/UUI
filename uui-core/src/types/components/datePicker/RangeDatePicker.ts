import { IHasRawProps } from '../../props';
import { BaseRangeDatePickerProps } from './BaseRangeDatePicker';

export type RangeDatePickerInputType = 'from' | 'to';

export interface RangeDatePickerCoreProps extends BaseRangeDatePickerProps {
    getPlaceholder?(type: RangeDatePickerInputType): string;
    rawProps?: {
        from?: IHasRawProps<React.HTMLAttributes<HTMLDivElement>>['rawProps'];
        to?: IHasRawProps<React.HTMLAttributes<HTMLDivElement>>['rawProps'];
        body?: IHasRawProps<React.HTMLAttributes<HTMLDivElement>>['rawProps'];
    };
}
