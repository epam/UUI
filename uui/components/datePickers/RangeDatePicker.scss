@use  '../../assets/styles/index' as *;

.dropdown-container {
    align-items: normal;
}

.container {
    display: flex;
    padding: 9px 30px;
    justify-content: space-between;
    border-top: 1px solid var(--uui-control-border);
    align-items: center;
    background-color: var(--uui-rdtp-bg);
    position: relative;
}

.button-group {
    display: flex;
    justify-content: flex-end;
}

.counter {
    color: var(--uui-control-border);
    font-size: 15px;
    font-family: var(--font-regular);
}

.date-input {
    min-width: 0;
    padding: 0 5px;

    &:global(.uui-focus) {
        border: 1px solid var(--uui-control-border-checked);
    }

    &.size-24 {
        @include control-box-layout(22px, 0);
    }

    &.size-30 {
        @include control-box-layout(28px, 0);
    }

    &.size-36 {
        @include control-box-layout(34px, 0);
    }

    &.size-42 {
        @include control-box-layout(40px, 0);
    }

    &.size-48 {
        @include control-box-layout(46px, 0);
    }
}

.date-input-group {
    display: flex;
    min-width: 0;
    background-color: var(--uui-rdtp-bg);
    border: 1px solid var(--uui-control-border);

    .separator {
        display: flex;
        flex-shrink: 0;
        background: var(--uui-control-border);
        width: 1px;
    }

    &:hover {
        border-color: var(--uui-control-textinput-border-hover);

        .separator {
            background: var(--uui-control-textinput-border-hover);
        }

        &:global(.uui-disabled), &:global(.uui-readonly) {
            border-color: var(--uui-control-border);

            .separator {
                background: var(--uui-control-border);
            }
        }
    }

    &:global(.uui-focus) {
        border-color: var(--uui-control-border);

        .separator {
            background: var(--uui-control-border-checked);
        }

        .date-input {
            &:hover {
                outline: 1px solid var(--uui-control-textinput-border-hover);
            }

            &:global(.uui-focus) {
                outline: 1px solid var(--uui-control-border-checked);
            }
        }
    }

    &:global(.uui-readonly) {
        @include readonly-input();

        .date-input {
            &:global(.uui-focus) {
                box-shadow: none;
            }
        }
    }

    &:global(.uui-invalid) {
        border-color: var(--uui-text-input-border-error);
    }
}
