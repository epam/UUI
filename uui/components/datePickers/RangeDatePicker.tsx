import * as React from 'react';
import cx from 'classnames';
import { IDropdownToggler, uuiMod, BaseRangeDatePickerProps, DropdownBodyProps } from "@epam/uui-core";
import { RangeDatePickerValue, BaseRangeDatePicker } from '@epam/uui-components';
import { DropdownContainer, FlexRow, SizeMod } from '../index';
import { RangeDatePickerBody } from './RangeDatePickerBody';
import { TextInput } from "../inputs";
import { systemIcons } from '../../icons/icons';
import { i18n } from '../../i18n';
import css from './RangeDatePicker.scss';

export interface RangeDatePickerProps extends BaseRangeDatePickerProps, SizeMod {
    getPlaceholder?(type: InputType): string;
}

export type InputType = 'from' | 'to';
const defaultValue: RangeDatePickerValue = { from: null, to: null };

export class RangeDatePicker extends BaseRangeDatePicker<RangeDatePickerProps> {
    renderBody(props: DropdownBodyProps) {
        return (
            <DropdownContainer { ...props } cx={ cx(css.dropdownContainer) }>
                <FlexRow>
                    <RangeDatePickerBody
                        cx={ cx(this.props.bodyCx) }
                        value={ this.getValue() }
                        onValueChange={ this.onRangeChange }
                        filter={ this.props.filter }
                        changeIsOpen={ this.toggleOpening }
                        presets={ this.props.presets }
                        focusPart={ this.state.inFocus }
                        renderDay={ this.props.renderDay }
                        renderFooter={ this.props.renderFooter && (() => this.props.renderFooter(this.props.value || defaultValue)) }
                        isHoliday={ this.props.isHoliday }
                        rawProps={ this.props.rawProps?.body }
                    />
                </FlexRow>
            </DropdownContainer>
        );
    }

    renderInput = (props: IDropdownToggler) => {
        return (
            <div
                className={ cx(
                    this.props.inputCx,
                    css.dateInputGroup,
                    this.props.isDisabled && uuiMod.disabled,
                    this.props.isReadonly && uuiMod.readonly,
                    this.props.isInvalid && uuiMod.invalid,
                    this.state.inFocus && uuiMod.focus,
                ) }
                onClick={ !this.props.isDisabled && props.onClick }
                onBlur={ this.handleWrapperBlur }
                ref={ props.ref }
            >
                <TextInput
                    icon={ systemIcons[this.props.size || '36'].calendar }
                    cx={ cx(css.dateInput, css['size-' + (this.props.size || 36)], this.state.inFocus === 'from' && uuiMod.focus) }
                    size={ this.props.size || '36' }
                    placeholder={ this.props.getPlaceholder ? this.props.getPlaceholder('from') : i18n.rangeDatePicker.pickerPlaceholderFrom }
                    value={ this.state.inputValue.from }
                    onValueChange={ this.getChangeHandler('from') }
                    isInvalid={ this.props.isInvalid }
                    isDisabled={ this.props.isDisabled }
                    isReadonly={ this.props.isReadonly }
                    onFocus={ (event) => this.handleFocus(event, 'from') }
                    onBlur={ (event) => this.handleBlur(event, 'from') }
                    isDropdown={ false }
                    rawProps={ this.props.rawProps?.from }
                />
                <div className={ css.separator } />
                <TextInput
                    cx={ cx(css.dateInput, css['size-' + (this.props.size || 36)], this.state.inFocus === 'to' && uuiMod.focus) }
                    placeholder={ this.props.getPlaceholder ? this.props.getPlaceholder('to') : i18n.rangeDatePicker.pickerPlaceholderTo }
                    size={ this.props.size || '36' }
                    value={ this.state.inputValue.to }
                    onCancel={ this.props.disableClear ? null : this.state.inputValue.from && this.state.inputValue.to && this.handleCancel }
                    onValueChange={ this.getChangeHandler('to') }
                    isInvalid={ this.props.isInvalid }
                    isDisabled={ this.props.isDisabled }
                    isReadonly={ this.props.isReadonly }
                    onFocus={ (e) => this.handleFocus(e, 'to') }
                    onBlur={ (e) => this.handleBlur(e, 'to') }
                    isDropdown={ false }
                    rawProps={ this.props.rawProps?.to }
                />
            </div>
        );
    }
}
