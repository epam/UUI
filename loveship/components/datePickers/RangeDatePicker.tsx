import * as React from 'react';
import { DropdownBodyProps, RangeDatePickerValue, BaseRangeDatePickerProps, BaseRangeDatePicker } from '@epam/uui-components';
import { DropdownContainer, FlexRow, TextInput, SizeMod, EditMode, RangeDatePickerBody } from '../index';
import { IDropdownToggler, uuiMod, IHasRawProps, cx } from '@epam/uui';
import { TextSettings } from '../../helpers/textLayout';
import { systemIcons } from '../icons/icons';
import { i18n } from '../../i18n';
import * as css from './RangeDatePicker.scss';

export interface RangeDatePickerProps extends BaseRangeDatePickerProps, SizeMod, TextSettings, EditMode {
    getPlaceholder?(type: InputType): string;
    rawProps?: {
        from?: IHasRawProps<HTMLDivElement>['rawProps'];
        to?: IHasRawProps<HTMLDivElement>['rawProps'];
        body?: IHasRawProps<HTMLDivElement>['rawProps'];
    }
}

export type InputType = 'from' | 'to';
const defaultValue: RangeDatePickerValue = { from: null, to: null };

export class RangeDatePicker extends BaseRangeDatePicker<RangeDatePickerProps> {
    renderBody(props: DropdownBodyProps) {
        return (
            <DropdownContainer cx={ this.props.cx }>
                <FlexRow>
                    <RangeDatePickerBody
                        value={ this.getValue() }
                        onValueChange={ this.onRangeChange }
                        filter={ this.props.filter }
                        changeIsOpen={ this.toggleOpening }
                        presets={ this.props.presets }
                        focusPart={ this.state.inFocus }
                        renderDay={ this.props.renderDay }
                        renderFooter={ this.props.renderFooter && (() => this.props.renderFooter(this.props.value || defaultValue)) }
                        isHoliday={ this.props.isHoliday }
                        rawProps={ this.props.rawProps?.body }
                    />
                </FlexRow>
            </DropdownContainer>
        );
    }

    renderInput = (props: IDropdownToggler) => {
        return (
            <div
                className={ cx(
                    css.dateInputGroup,
                    css[`date-input-group-${this.props.mode}`],
                    this.props.isDisabled && uuiMod.disabled,
                    this.props.isReadonly && uuiMod.readonly,
                    this.props.isInvalid && uuiMod.invalid,
                ) }
                onClick={ props.onClick }
                onBlur={ this.handleWrapperBlur }
            >
                <TextInput
                    icon={ systemIcons[this.props.size || '36'].calendar }
                    cx={ cx(css.dateInput, css['size-' + (this.props.size || 36)], this.state.inFocus === 'from' && uuiMod.focus) }
                    mode={ this.props.mode || 'form' }
                    size={ this.props.size || '36' }
                    lineHeight={ this.props.lineHeight }
                    fontSize={ this.props.fontSize }
                    placeholder={ this.props.getPlaceholder ? this.props.getPlaceholder('from') : i18n.rangeDatePicker.pickerPlaceholderFrom }
                    value={ this.state.inputValue.from }
                    onValueChange={ this.getChangeHandler('from') }
                    isInvalid={ this.props.isInvalid }
                    isDisabled={ this.props.isDisabled }
                    isReadonly={ this.props.isReadonly }
                    onFocus={ () => this.handleFocus('from') }
                    onBlur={ () => this.handleBlur('from') }
                    isDropdown={ false }
                    rawProps={ this.props.rawProps?.from }
                />
                <div className={ css.separator } />
                <TextInput
                    cx={ cx(css.dateInput, css['size-' + (this.props.size || 36)], this.state.inFocus === 'to' && uuiMod.focus) }
                    placeholder={ this.props.getPlaceholder ? this.props.getPlaceholder('to') : i18n.rangeDatePicker.pickerPlaceholderTo }
                    mode={ this.props.mode || 'form' }
                    size={ this.props.size || '36' }
                    lineHeight={ this.props.lineHeight }
                    fontSize={ this.props.fontSize }
                    value={ this.state.inputValue.to }
                    onCancel={ this.props.disableClear ? null : this.state.inputValue.from && this.state.inputValue.to && this.handleCancel }
                    onValueChange={ this.getChangeHandler('to') }
                    isInvalid={ this.props.isInvalid }
                    isDisabled={ this.props.isDisabled }
                    isReadonly={ this.props.isReadonly }
                    onFocus={ () => this.handleFocus('to') }
                    onBlur={ () => this.handleBlur('to') }
                    isDropdown={ false }
                    ref={ (el) => this.toTextInput = el } /* to make the first picker to be the target of dropdown */
                    rawProps={ this.props.rawProps?.to }
                />
            </div>
        );
    }
}