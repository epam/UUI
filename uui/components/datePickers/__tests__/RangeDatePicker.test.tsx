import * as React from 'react';
import { RangeDatePicker, RangeDatePickerProps } from '../RangeDatePicker';
import { renderSnapshotWithContextAsync, setupComponentForTest, fireEvent, screen, within } from '@epam/test-utils';

jest.mock('react-popper', () => {
    const PopperJS = jest.requireActual('react-popper');
    const Popper = function ({ children }: any) {
        return children({
            ref: jest.fn,
            placement: 'bottom-start',
            style: {
                position: 'fixed',
                top: 0,
                right: 'auto',
                bottom: 'auto',
                left: 0,
            },
            update: jest.fn(),
            isReferenceHidden: false,
            arrowProps: {
                ref: jest.fn,
            },
        });
    };
    return {
        ...PopperJS,
        Popper,
    };
});

async function setupRangeDatePicker(params: { value: { from: string, to: string } | null, format?: string }) {
    const { value, format } = params;

    const { result, mocks } = await setupComponentForTest<RangeDatePickerProps>(
        (context) => ({
            rawProps: { from: { 'data-testid': 'from' }, to: { 'data-testid': 'to' } },
            value,
            format,
            onValueChange: jest.fn().mockImplementation((newValue) => {
                context.current.setProperty('value', newValue);
            }),
        }),
        props => <RangeDatePicker { ...props } />,
    );

    const from = within(result.getByTestId('from')).getByRole('textbox') as HTMLInputElement;
    const to = within(result.getByTestId('to')).getByRole('textbox') as HTMLInputElement;
    const clear = result.container.querySelector('.uui-icon-cancel');

    return {
        result,
        dom: { from, to, clear },
        mocks: { onValueChange: mocks.onValueChange },
    };
}

describe('RangeDataPicker', () => {
    it('should be rendered if minimum params and custom format defined', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <RangeDatePicker
                format="MMM D, YYYY"
                value={ null }
                onValueChange={ jest.fn }
            />,
        );
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered if many params defined', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <RangeDatePicker
                format="MMM D, YYYY"
                value={ null }
                onValueChange={ jest.fn }
                renderFooter={ ((value: any) => jest.fn(value)) as any }
                disableClear={ false }
                getPlaceholder={ (type) => '' }
                isDisabled
                isReadonly
                isInvalid
            />,
        );
        expect(tree).toMatchSnapshot();
    });

    it('should change input value after change props', async () => {
        const value = { from: '2017-01-22', to: '2017-01-28' };
        const { dom } = await setupRangeDatePicker({ value });
        expect(dom.from.value).toBe('Jan 22, 2017');
        expect(dom.to.value).toBe('Jan 28, 2017');
    });

    it('should render with default props', async () => {
        const { result } = await setupRangeDatePicker({ value: null });
        expect(result.container).not.toBeFalsy();
    });

    it('should change state on picker clear', async () => {
        const value = { from: '2017-01-22', to: '2017-01-28' };
        const { dom, mocks } = await setupRangeDatePicker({ value });
        fireEvent.click(dom.clear);
        expect(mocks.onValueChange).toBeCalledWith({ from: null, to: null });
    });

    it(`should open picker on 'from' field focus and close it on blur`, async () => {
        const { dom } = await setupRangeDatePicker({ value: null });
        fireEvent.focus(dom.from);
        expect(await screen.queryByRole('dialog')).toBeTruthy();
        fireEvent.blur(dom.from);
        expect(await screen.queryByRole('dialog')).toBeFalsy();
    });

    it(`should open picker on "To" field focus and close it on blur`, async () => {
        const { dom } = await setupRangeDatePicker({ value: null });
        fireEvent.focus(dom.to);
        expect(await screen.queryByRole('dialog')).toBeTruthy();
        fireEvent.blur(dom.to);
        expect(await screen.queryByRole('dialog')).toBeFalsy();
    });

    it('should reset invalid "From" value onBlur', async () => {
        const value = { from: '2019-10-47', to: '2019-10-07' };
        const { dom, mocks } = await setupRangeDatePicker({ value });
        fireEvent.blur(dom.from);
        expect(mocks.onValueChange).toHaveBeenLastCalledWith({
            from: null,
            to: value.to,
        });
    });

    it('should set new value when new value typed in input', async () => {
        const newValueManualTyping = { from: 'Sep 11, 2019', to: 'Sep 20, 2019' };
        const value = { from: '2019-09-14', to: '2019-09-15' };
        const { dom, mocks } = await setupRangeDatePicker({ value });

        fireEvent.change(dom.from, { target: { value: newValueManualTyping.from } });
        fireEvent.blur(dom.from);
        expect(mocks.onValueChange).toHaveBeenLastCalledWith({
            from: '2019-09-11',
            to: '2019-09-15',
        });
    });

    it('should change range on picker body value change', async () => {
        const newValueManualSel = { from: '2019-09-11' };
        const value = { from: '2019-09-10', to: '2019-09-12' };
        const { dom, mocks } = await setupRangeDatePicker({ value });

        fireEvent.focus(dom.from);
        const dialog = await screen.queryByRole('dialog');
        const [sept11, oct11] = await within(dialog).findAllByText('11');
        fireEvent.click(sept11);
        expect(mocks.onValueChange).toHaveBeenLastCalledWith({
            from: newValueManualSel.from,
            to: '2019-09-12',
        });
    });

    it('should format value onBlur', async () => {
        const value = { from: '2019-09-10', to: '2019-10-10' };
        const newValueInIsoFormat = { from: '2019-09-11', to: '2019-10-11' };

        const { dom } = await setupRangeDatePicker({ value });
        expect(dom.from.value).toBe('Sep 10, 2019');
        fireEvent.change(dom.from, { target: { value: newValueInIsoFormat.from } });
        fireEvent.blur(dom.from);
        expect(dom.from.value).toBe('Sep 11, 2019');

        fireEvent.change(dom.to, { target: { value: newValueInIsoFormat.to } });
        fireEvent.blur(dom.to);
        expect(dom.to.value).toBe('Oct 11, 2019');
    });

    it('should return format', async () => {
        const value = { from: '2019-09-10', to: '2019-10-10' };
        const { dom } = await setupRangeDatePicker({ value, format: 'DD-MM-YYYY' });
        expect(dom.from.value).toBe('10-09-2019');
        expect(dom.to.value).toBe('10-10-2019');
    });

    it('should set the same value on from: & to: input', async () => {
        const value = { from: '2019-09-10', to: '2019-09-10' };
        const { dom } = await setupRangeDatePicker({ value });
        expect(dom.from.value).toBe('Sep 10, 2019');
        expect(dom.to.value).toBe('Sep 10, 2019');
    });
});
