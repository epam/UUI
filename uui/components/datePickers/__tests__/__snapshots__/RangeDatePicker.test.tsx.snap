// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeDataPicker should be rendered if many params defined 1`] = `
<div
  className="dateInputGroup uui-disabled uui-readonly uui-invalid"
  onBlur={[Function]}
  onClick={false}
>
  <div
    className="container uui-input-box uui-disabled uui-readonly uui-invalid root size-36 mode-form dateInput size-36"
    onBlur={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <div
      className="container uui-icon uui-enabled"
      style={Object {}}
    >
      <svg
        className=""
      />
    </div>
    <input
      aria-disabled={true}
      aria-invalid={true}
      aria-readonly={true}
      className="uui-input"
      disabled={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      readOnly={true}
      tabIndex={-1}
      type="text"
      value=""
    />
  </div>
  <div
    className="separator"
  />
  <div
    className="container uui-input-box uui-disabled uui-readonly uui-invalid root size-36 mode-form dateInput size-36"
    onBlur={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <input
      aria-disabled={true}
      aria-invalid={true}
      aria-readonly={true}
      className="uui-input"
      disabled={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      readOnly={true}
      tabIndex={-1}
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`RangeDataPicker should be rendered if minimum params and custom format defined 1`] = `
<div
  className="dateInputGroup"
  onBlur={[Function]}
  onClick={[Function]}
>
  <div
    className="container uui-input-box -clickable root size-36 mode-form dateInput size-36"
    onBlur={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <div
      className="container uui-icon uui-enabled"
      style={Object {}}
    >
      <svg
        className=""
      />
    </div>
    <input
      className="uui-input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="From:"
      tabIndex={0}
      type="text"
      value=""
    />
  </div>
  <div
    className="separator"
  />
  <div
    className="container uui-input-box -clickable root size-36 mode-form dateInput size-36"
    onBlur={[Function]}
    onFocus={[Function]}
    tabIndex={-1}
  >
    <input
      className="uui-input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="To:"
      tabIndex={0}
      type="text"
      value=""
    />
  </div>
</div>
`;
